### PR TITLE
GAWB-1419: in datasets table, show whether you have access to workspace for a dataset

### PR DESCRIPTION
**Pull request by skwalker**
in conjunction with https://github.com/broadinstitute/firecloud-orchestration/pull/391

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [ ] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/library/library_page.cljs [hunk 1]
```diff
@@ -74,65 +81,63 @@
...
-                (if success?
-                  (let [{:keys [total results aggregations]} (get-parsed-response)]
-                    (swap! state assoc :total total)
-                    (callback {:group-count total
-                               :filtered-count total
-                               :rows results})
-                    (when (= 1 current-page)
-                      ((:callback-function props) aggregations)))
-                  (callback {:error status-text})))})))))})
+          (let [from (* (- current-page 1) rows-per-page)]
```
Comment (skwalker) [inline src/cljs/main/broadfcui/page/library/library_page.cljs:121]: ugh I always change the spacing of the entire page without meaning to... but this should be right? it's what intellij gave me..

Comment (MattPutnam) [reply src/cljs/main/broadfcui/page/library/library_page.cljs:121]: Yes, this one is correct.

**Review by davidangb — APPROVED** [review]
👍 code looks good to me as TL. My only concern is with the UX and exact wording - but that'll get worked out via PO review.

**Review by davidangb — APPROVED** [review]
👍

**Review by rushtong — APPROVED** [review]
👍 Looks good, works good! Just need to work out the final language for the hover text.

File: src/cljs/main/broadfcui/page/library/library_page.cljs [hunk 2]
```diff
@@ -59,11 +59,13 @@
          :cell-content-style {:padding nil}
          :columns (concat
                    [{:resizable? false :width 30 :reorderable? false
-                     :as-text (fn [access]
-                                (if (= access "NO ACCESS") "You don't have access to the workspace for this dataset.")) ;; what should the wording for this be?
-                     :content-renderer (fn [access]
-                                         (if (= access "NO ACCESS")
-                                           (icons/icon {:style {:alignSelf "center"}} :shield)))}
+                     :as-text (fn [data]
+
```
Comment (rushtong) [inline src/cljs/main/broadfcui/page/library/library_page.cljs]: Extra line

File: src/cljs/main/broadfcui/page/library/library_page.cljs [hunk 3]
```diff
@@ -82,7 +84,7 @@
                        (concat [:library:indication :library:dataUseRestriction :library:numSubjects])
                        (map data)
                        (cons data)
-                       (cons (:workspaceAccess data))))}]))
```
Comment (rushtong) [inline src/cljs/main/broadfcui/page/library/library_page.cljs]: Nice simplification - this row function twists my brain unnaturally.

**Comment by rushtong** [comment]
👍

**Review by davidangb — APPROVED** [review]
👍 for last two commits

**Pull request merged**